src/components/FilePreview.js

```diff
@@ -32,9 +32,18 @@ export default class FilePreview extends Component {
     }
     return (
       <div className="file-preview">
+        {
+          file.from_theme &&
+            <span className="theme-indicator">
+              <i className="fa fa-diamond" aria-hidden="true" title="Theme Asset" />
+            </span>
+        }
         {nodeLink}
         <span className="filename">{file.path}</span>
-        <button onClick={() => this.handleClickDelete(file.path)} className="delete" title="Delete file">x</button>
+        {
+          !file.from_theme &&
+            <button onClick={() => this.handleClickDelete(file.path)} className="delete" title="Delete file">x</button>
```

Comment:
> Why should we not allow deleting theme files?

While I was greatly surprised that you asked this question, on reflecting further, I'm assuming you were talking about removing the `Jekyll::StaticFile` object from the `site_payload` rather than deleting the file proper..

To answer otherwise, three things to consider:
 - The theme-gem is supposed to be a read-only object for Jekyll.
 - We have no write-access outside `site.source`.
 - When users click `delete`, on a theme-file they're sending a `delete` request to the `relative_path` of the static file which is sanitized to point to a non-existing entity in the `source_dir`. (e.g. `<theme>/assets/logo.png` will resolve as `<site.source>/assets/logo.png` which doesn't exist. Otherwise, Jekyll would've read that file in to memory instead)

There's no valid use-case to allow write-access to the theme-gem in future either